docs/sections/user_guide/yaml/components/make_hgrid.rst

```diff
@@ -0,0 +1,252 @@
+.. _make_hgrid_yaml:
+
+make_hgrid
+==========
+
+Structured YAML to run :ufs-utils:`make_hgrid<make-hgrid>` is validated by JSON Schema and requires the ``make_hgrid:`` block, described below. If ``make_hgrid`` is to be run via a batch system, the ``platform:`` block, described :ref:`here <platform_yaml>`, is also required.
+
+Here is a prototype UW YAML ``make_hgrid:`` block, explained in detail below:
+
+.. highlight:: yaml
+.. literalinclude:: ../../../../shared/make_hgrid.yaml
+
+UW YAML for the ``make_hgrid:`` Block
+-------------------------------------
+
+execution:
+^^^^^^^^^^
+
+See :ref:`here <execution_yaml>` for details.
+
+run_dir:
+^^^^^^^^
+
+The path to the directory where ``make_hgrid`` will write its outputs.
+
+config:
+^^^^^^^
```

Comment:
Any reason not to have the items under `config:` and in the corresponding `make_hgrid.yaml` file sorted? Maybe there's a reason for the current order, but I couldn't guess it.

Reply:
I've listed them in order of the `make_hgrid` function docs, but I can resort them to something actually reasonable!

Reply:
I'm somewhat inclined to put grid_type first as it defines the subsequent parameters, and maintaining the similar parameter groupings like nx/ny - then most of the alphabetical order is already broken, but I can improve some of it

Reply:
I think that'd be good if accompanied by a little additional text stating e.g. that `grid_type` is kind of a primary choice, and that the options that follow depend on that choice. In fact, is `grid_type` perhaps even required? Does it make sense not to have it?

Maybe we could present a table visually mapping `grid_type` values to the options that then become required for and are related to them.

Reply:
As reluctant as I am to update the jsonschema and its test again, I think that might be the way to go.

Reply:
AH, it is already required - I'm starting to lose track of all the conditions! I'll think about a visual map

Reply:
I've made sure all the key names have double-backstrokes and have used the table for `grid_type`, this helps a lot without adding further diagrams here